### PR TITLE
#169007986 Restructuring code base for maintainability 

### PR DESCRIPTION
**Pull request by jptuyisenge**
#### What does this PR do?
Thi PR does the following :
 - Restructure codebase for maintainability
 - Make your codebase modular
 - Remove repetitions of function (use function once)
 - Create a helper function to handle all returns.
 - Remove password returns in all situation

#### Description of Task to be completed?
 - Restructure codebase for maintainability
 - Make your codebase modular
 - Remove repetitions of function (use function once)
 - Create a helper function to handle all returns.
 - Remove password returns in all situation

#### How should this be manually tested?
Use Postman to test all endpoints;
Key: Content-Type value: application/JSON
Test endpoints

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169007986](https://www.pivotaltracker.com/story/show/169007986)
#### Screenshots (if appropriate)
N/A
image
image

#### Questions:
N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "express": "^4.17.1",
     "joi": "^14.3.1",
     "jsonwebtoken": "^8.5.1",
+    "lodash": "^4.17.15",
     "mocha": "^6.2.0",
     "nodemon": "^1.19.2",
     "nyc": "^14.1.1",
```
Comment (Oluwasegun-AA) [inline package.json:34]: @tuyisengepaul , lock your dependencies and dev-dependencies by removing all caret(^) symbols before the dependency version. this ensure that the same version of the module is installed whichever year you return to this project. hence it attracts no breaking changes due to file upgrade

File: server/controllers/articles.js [hunk 2]
```diff
@@ -35,11 +24,7 @@ class allAboutArticle {
       flag: 0,
     };
     articles.push(data);
-    return res.status(201).json({
-      status: '201',
-      message: 'article successfuly created',
-      data,
-    });
+    returnFc(req, res, '201', 'article successfuly created', data);
```
Comment (Oluwasegun-AA) [inline server/controllers/articles.js:27]: @tuyisengepaul  i do not understand what you mean by `returnFc()` please make your functions more descriptive

**Review by Oluwasegun-AA — CHANGES_REQUESTED** [review]

File: server/controllers/articles.js (hunk 2 above)
Comment (jptuyisenge) [reply server/controllers/articles.js:27]: This is the helper function to handle all responses.

File: package.json (hunk 1 above)
Comment (jptuyisenge) [reply package.json:34]: Resolved

File: server/controllers/articles.js (hunk 2 above)
Comment (jptuyisenge) [reply server/controllers/articles.js:27]: Resolved

**Pull request merged**